libmariadb/CMakeLists.txt

```diff
@@ -4,6 +4,7 @@ INCLUDE_DIRECTORIES(${CC_SOURCE_DIR}/include
 ADD_DEFINITIONS(-D HAVE_COMPRESS)
 ADD_DEFINITIONS(-D LIBMARIADB)
 ADD_DEFINITIONS(-D THREAD)
+ADD_OPTION(BUILD_SHARED_LIBS "Build as a shared library" ON)
```

Comment:
I don't think thats a good idea. If you don't need a shared library, just don't use it (or remove) it. The overhead for building shared/static  libraries is minimal, since they are created from object library.

Reply:
@9EOR9 thans for the reply.

can you give me more details please? you mean to remove t he default ON for this option?

or you mean that this change is not needed to achieve what im looking?

my context its:

- im building an backend application that use mariadb connector, 
- that app is build as static 
- my binary need to be able to work alone without need the shared libs
- the the binary is moved a lives inside alpine docker image that don't have any other deps than my binary, that means the mariadb lib is not in the system as a shared lib
- that alpine docker image is used on gcloud run

If mariadb is not build as static and linked to my binary, when I execute my binary, I have the Undefined Reference error.

having mariadb build statically and linking the .a library with my binary, not an issue.

Reply:
>that means the mariadb lib is not in the system as a shared lib

actually that doesn't. You shouldn't necessarily put a shared lib in a system folder. All you need is specify LD_LIBRARY_PATH for the app to run (or ldconfig it). Anyway, if you don't need a shared object, you can just rm it with our packaging script -- I guess that was @9EOR9's point.

Reply:
Thanks for your response.  But I need to put that shared library on the docker alpine also right? I mean the libmaria…so 

I’m looking for a way to be able to build my binary statically with libmariadb and don’t need to have the shared library on the docker alpine image. 

The way that I found was build libmaria statically and then link with my binary, by apply the changes from this PR.

so my intention is just to have a way to build Mariadb as static library

Reply:
Static libmariadb is named mariadbclient, and it's always built.
ADD_LIBRARY(mariadbclient STATIC ${MARIADB_OBJECTS} ${EMPTY_FILE})
You don't it need additionally built under another name